shopping_mall/backend/ai/agent/supervisor/executor.py

```
@@ -0,0 +1,426 @@
+
+            # 진행 중인 플로우가 있어도, 사용자가 명시적으로 다른 의도를 표현하면 새 플로우 시작
+            pending_action = snapshot.values.get("action") if snapshot.next else None
+            intent_mismatch = (
+                pending_action is not None
+                and pending_action != new_action
+                and any(kw in query for kw in _EXCHANGE_KEYWORDS_ACTION | _CANCEL_KEYWORDS_ACTION)
+            )
+
+            if snapshot.next and not intent_mismatch:
+                # 진행 중인 플로우 재개
+                logger.info(f"[order_graph] 플로우 재개 — session={session_id} query='{query[:60]}'")
```

Ruff (0.15.10)

[warning] Logging statement uses f-string

(G004)